packages/flutter/lib/src/widgets/_web_image_web.dart

```diff
@@ -0,0 +1,241 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:js_interop';
+import 'dart:ui_web' as ui_web;
+
+import '../../rendering.dart';
+import '../../widgets.dart';
+import '../painting/_network_image_web.dart';
+import '../web.dart' as web;
+
+/// Returns [true] if the bytes at [url] can be fetched.
+///
+/// The bytes may be unable to be fetched if they aren't from the same origin
+/// and the server hosting them does not allow cross-origin requests.
+Future<bool> checkIfImageBytesCanBeFetched(String url) async {
+  // First check if url is same origin.
+  final Uri uri = Uri.parse(url);
+  if (uri.origin == web.window.origin) {
+    return true;
```

Comment:
Can we return a `SynchronousFuture` in this case, so that in case of this "technically" synchronous check, the widget does not have to wait one frame for the event loop to resolve the result?

Reply:
Done.